packages/contracts-bedrock/test/Authorization.t.sol

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import { CommonTest } from "test/setup/CommonTest.sol";
+import { Executables } from "scripts/Executables.sol";
+import { console2 as console } from "forge-std/console2.sol";
+import { ProtocolVersions } from "src/L1/ProtocolVersions.sol";
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+
+/// @title Authorization_Test
+/// @dev Specify access requirements of all entrypoints to L1 contracts.
+///      New contract and methods should be added to the `setUp` function.
+contract Authorization_Test is CommonTest {
+    struct AbiEntry {
+        string fnName;
+        bytes4 sel;
+    }
+
+    struct Abi {
+        string contractName;
+        AbiEntry[] entries;
+    }
+
+    struct Spec {
+        string name;
+        bytes4 sel;
+        bool auth;
```

Comment:
@mds1 For now, each `Spec` documents which methods can only be used by contract owners. What other types of auth are worth documenting? I could see maybe "pauseability" going in here in the future.

Reply:
I like how this file is setup overall, and specifying pausability status makes sense to me also, but cc @maurelian since I know we were discussing that the other day.

The only downside to the current test approach is we don't test that the implementations match the `Spec`'s auth bool. For `Initializeable` this was easy to do because it's small amount of functions and initialization is the first thing checked during execution so calldata doesn't matter. It's a lot harder to do something like in the general case for all functions here, so I think having this act just as a length check + forcing function for the spec is ok (pending a better solution, ref your notion [doc](https://www.notion.so/oplabs/Contract-Specifications-617716b8d19a44b381038d59cbde3c1e?pvs=4))).

Reply:
+1 to adding pausability on this, but I think it's out of scope for this PR.

Reply:
@Inphi What do you think of changing `bool auth` to `Role auth[]` where:
```solidity
enum Role {
  Challenger,
  L1ProxyAdminOwner,
  L2ProxyAdminOwner,
  SystemConfigOwner,
  MintManagerOwner,
  Guardian
}
```
Roles sourced from this [notion doc](https://www.notion.so/oplabs/4ab895de0c36475ab3ef292cf12adc00?v=511aead484f54a06b89fb2672172611d&pvs=4), where MSD Controller is outdated. cc @maurelian for thoughts / confirmation on roles.

If we like this approach for more granular specificity I'd suggest pulling it into a separate issue to avoid blocking this PR

Reply:
> If we like this approach for more specificity I'd suggest pulling it into a separate issue to avoid blocking this PR

💯 to doing this. 
Unopinionated about doing it in this PR or separately.

Reply:
Created an issue to track here: https://github.com/ethereum-optimism/client-pod/issues/308

Pulling out to a separate issue seems ideal so we can unblock you to add `paused` status + unblock @Inphi to work on the state diff bug :)

Reply:
I'll leave this to be done on another PR. Thanks